integration-test/src/test/java/com/datastrato/gravitino/integration/test/container/BaseContainer.java

```diff
@@ -124,7 +125,33 @@ public void start() {
     container.start();
   }
 
-  protected abstract boolean checkContainerStatus(int retryLimit);
+  protected abstract boolean checkContainerStatus(int timeoutMillis);
+
+  protected boolean checkContainerStatusWithRetry(int timeoutMillis, BooleanSupplier... checker) {
+    int nRetry = 0;
+    int sleepTimeMillis = 5_000;
+    int totalSleepTimeMillis = 0;
+
+    // retry until timeout
+    for (BooleanSupplier check : checker) {
+      while (totalSleepTimeMillis < timeoutMillis) {
+        if (check.getAsBoolean()) {
+          break;
+        }
+
+        sleepTimeMillis = sleepTimeMillis * (int) Math.pow(2, nRetry++);
+        sleepTimeMillis = Math.min(sleepTimeMillis, timeoutMillis - totalSleepTimeMillis);
```

Comment:
`timeoutMillis - totalSleepTimeMillis` Will there be a negative number?

Reply:
No, Since the loop condition is `totalSleepTimeMillis < timeoutMillis`, the expression `timeoutMillis - totalSleepTimeMillis` will always be greater than 0 in this code block.